### PR TITLE
add helpUsImprove field to the backend

### PR DESCRIPTION
**Pull request by kdachev97**
This ticket is related to the frontend ticket: feature/add-tickbox-to-registration-field

Added the helpUsImprove field to the schema.prisma and register.dto.ts files as requested

### CODE REVIEW

File: schema.prisma [hunk 1]
```diff
@@ -40,6 +40,7 @@ model Person {
   updatedAt          DateTime?           @updatedAt @map("updated_at") @db.Timestamptz(6)
   // Receive marketing notifications
   newsletter         Boolean?            @default(false)
+  helpUsImporve      Boolean?            @default(false)
```
Comment (sashko9807) [inline schema.prisma:43]: There is a typo. Change to `helpUsImprove`
We use snake_case for referencing a field via raw SQL query. Would you please add @map("help_us_improve") at the end of this line. 
Example:
```js
helpUsImporve      Boolean?            @default(false) @map("help_us_improve")
```

**Review by sashko9807 — CHANGES_REQUESTED** [review]
Great job!
When changing the prisma schema in order for those changes to be implemented in the database we need to run `yarn prisma migrate dev`.  This command will automatically generate the SQL query to add the help_us_improve field to the persons table, as well as auto-generating the changes for the Prisma Client.

Once `yarn prisma migrate dev` is finished, execute `yarn format`, which will run prettier on all generated.
 
So in summary:
1. Run `yarn prisma migrate dev` to generate the SQL query which updates the person table 
2. Run `yarn format`  - to prettify the generated Prisma clients.
3. Commit the changes.

If you have further questions regarding the requested changes, feel free to contact me.

**Review by sashko9807 — APPROVED** [review]

**Comment by kdachev97** [comment]
Should I delete the previous migration?

На пн, 18.03.2024 г. в 14:01 Aleksandar Petkov ***@***.***>
написа:

**Comment by sashko9807** [comment]
> Should I delete the previous migration?
It would be best if you can. As it will keep the migration folder more tidier.


Please fix the typo, as well as add `@map("help_us_improve")`, as we use snake_case for referencing db fields directly.
Example: 
```js
helpUsImprove      Boolean?            @default(false) @map("help_us_improve")
```
Then re-execute the `yarn prisma migrate dev` and `yarn format`
Once the commands from above are executed, please also fix the failing API test job due to missing helpUsImprove field in [personMock.ts](https://github.com/podkrepi-bg/api/blob/master/apps/api/src/person/__mock__/peronMock.ts), e.g. just add `helpUsImprove` at the end of the object.

**Comment by sashko9807** [comment]
Hmm. apparently I forgot to tell you to add the following line:
```js
helpUsImprove: registerDto.helpUsImprove ? registerDto.helpUsImprove : false
``` 
to the person.upsert method [here](https://github.com/podkrepi-bg/api/blob/master/apps/api/src/auth/auth.service.ts#L295). Do yo mind adding it? Sorry for that.

**Comment by kdachev97** [comment]
Do I need to run new migrations or is it just that?

На пн, 18.03.2024 г. в 19:19 Aleksandar Petkov ***@***.***>
написа:

**Comment by sashko9807** [comment]
> Do I need to run new migrations or is it just that?

Just that change. Everything regarding the migration is OK.

**Review by sashko9807 — APPROVED** [review]
